src/subprocess.rs

```diff
@@ -122,7 +232,42 @@ impl SubProcessIntf for SubProcess {
             sub_proc = sub_proc.env(key, value);
         }
 
-        sub_proc.join()?;
+        if let Err(err) = sub_proc.join() {
```

Comment:
This is fine. You could also implement the `From` trait to handle typo conversion between two `Error` types as well.

Reply:
Originally, I could not get the `From` trait  to work because the underlying `PopenError` type had not implemented the `Clone` trait. I now see that the `From` trait implementation can convert the `PopenError` into a String and avoid the need to implement the `Clone` trait for `PopenError`.